app/models/concerns/waste_exemptions_engine/can_deactivate_exemption.rb

```diff
@@ -15,19 +15,27 @@ module CanDeactivateExemption
 
         event :cease do
           transitions from: :active,
-                      to: :ceased
+                      to: :ceased,
+                      after: :deregister_exemption
         end
 
         event :revoke do
           transitions from: :active,
-                      to: :revoked
+                      to: :revoked,
+                      after: :deregister_exemption
         end
 
         event :expire do
           transitions from: :active,
                       to: :expired
         end
       end
+
+      # Transition effects
+      def deregister_exemption
```

Comment:
I wonder if a name like `update_deregistered_on` would be a bit clearer? My initial assumption of what the method does based on this name would be that it was applying the state change.

Reply:
Fixed in 3c6532f.